python/tskit/trees.py

```diff
@@ -2370,6 +2370,67 @@ def kc_distance(self, other, lambda_=0.0):
         """
         return self._ll_tree.get_kc_distance(other._ll_tree, lambda_)
 
+    def split_polytomies(
+        self,
+        *,
+        epsilon=None,
+        method=None,
+        record_provenance=True,
+        random_seed=None,
+    ):
+        """
+        Return a new :class:`.Tree` where extra nodes and edges have been inserted
+        so that any any node ``u`` with greater than 2 children --- a multifurcation
+        or "polytomy" --- is resolved into successive bifurcations. New nodes are
+        inserted at times fractionally less than than the time of node ``u``
+        (controlled by the ``epsilon`` parameter).
+
+        If the ``method`` is ``"random"`` (currently the only option, and the default
+        when no method is specified), then for a node with :math:`n` children, the
+        :math:`(2n - 3)!!` possible bifurcating topologies are produced with equal
+        probability.
+
+        .. fixme:: Looks like this function is wrong
+
+        The returned :class`.Tree` will have the same genomic span as this tree,
+        and node IDs will be converved (that is, node ``u`` in this tree will
+        also exist in the returned tree). The returned tree is derived from a
+        tree sequence that contains only one non-degenerate tree, that is, where
+        edges cover only the interval spanned by this tree.
+
+        .. note::
+            A tree sequence :ref:`requires<sec_valid_tree_sequence_requirements>` that
+            parents be older than children and that mutations are younger than the
+            parent of the edge on which they lie. If ``epsilon`` is not small enough,
+            compared to the distance between a polytomy and its oldest child (or oldest
+            child mutation) these requirements may not be met. In this case an error is
+            raised, recommending a smaller epsilon value be used.
```

Comment:
An alternative to epsilon would be to use https://numpy.org/doc/stable/reference/generated/numpy.nextafter.html. That should give finer resolution than trying to add a small floating point value.

Reply:
I didn't know about nextafter. This is nice, but I'm cautious about using it, because we have been bitten in the past by creating nodes fractionally smaller than something, and then not being able to wedge in extra stuff later on (e.g. mutations). I can't see why you might want to do this in this case, but I'm still wary. Anyway, Jerome's suggestion of defaulting to an even spread of times seems good to me.

Reply:
I agree with Yan here - if folks want to have a super-fine epsilon then they can provide it as a parameter, but as a default nextafter is a bit tricky. In practise 1e-10 is a reasonable default, I think, leaving enough space in between nodes for mutations (if they are needed) while still being very small.

I've changed my mind on the auto-spacing after thinking more about it @hyanwong - it'll be quite hard to test, and actually you probably *do* want super short branches on the randomly resolved polytomies.